clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -87,13 +90,36 @@
                                                                         "Metadata for each partition of the topic."));
 
     public static final Schema METADATA_RESPONSE_V0 = new Schema(new Field("brokers",
-                                                                           new ArrayOf(BROKER),
+                                                                           new ArrayOf(BROKER_V0),
                                                                            "Host and port information for all brokers."),
                                                                  new Field("topic_metadata",
                                                                            new ArrayOf(TOPIC_METADATA_V0)));
 
-    public static final Schema[] METADATA_REQUEST = new Schema[] {METADATA_REQUEST_V0};
-    public static final Schema[] METADATA_RESPONSE = new Schema[] {METADATA_RESPONSE_V0};
+    public static final Schema BROKER_V1 = new Schema(new Field("node_id", INT32, "The broker id."),
```

Comment:
I am of the impression that we are not consolidating so that the wire protocols can change independently. However, we want to represent that in Java or Scala object is up to the parser. This decouples protocol from implementation. Though even in implementation we have chosen to duplicate in the past too.